documentation-website/Writerside/topics/DAO-CRUD-Operations.topic

```diff
@@ -0,0 +1,254 @@
+                Once the query is defined, you can wrap the result in the <code>User</code> entity using the
+                <code>wrapRows()</code> function to generate entities from the retrieved data:</p>
+            <code-block lang="kotlin">
+                val users = User.wrapRows(query).toList()
+            </code-block>
+        </chapter>
+
+    </chapter>
+    <chapter title="Update" id="update">
+        <p>
+            You can update the value of a property just as you would with any property in a Kotlin class:
+        </p>
+        <code-block lang="kotlin">
+            movie.name = &quot;Episode VIII – The Last Jedi&quot;
+        </code-block>
+        <note>
+            Exposed doesn't make an immediate update when you set a new value for <code>Entity</code>, it just stores it on the inner map.
+            "Flushing" values to the database occurs at the end of the transaction, or before the next <code>SELECT *</code> from the database.
+        </note>
+        <chapter title="findByIdAndUpdate" id="findByIdAndUpdate">
+            <p>
+                To search for an entity by its id and apply an update, use the <code>findByIdAndUpdate()</code> function:
+            </p>
+            <code-block lang="kotlin">
+                val updatedMovie = StarWarsFilm.findByIdAndUpdate(5) {
+                    it.name = &quot;Episode VIII – The Last Jedi&quot;
+                }
+            </code-block>
+        </chapter>
+        <chapter title="findSingleByAndUpdate" id="findSingleByAndUpdate">
+            <p>
+                To search for a single entity by a query and apply an update, use the <code>findSingleByAndUpdate()</code>
+                function:
+            </p>
+            <code-block lang="kotlin">
+                val updatedMovie2 = StarWarsFilm.findSingleByAndUpdate(StarWarsFilms.name eq &quot;The Last Jedi&quot;) {
+                    it.name = &quot;Episode VIII – The Last Jedi&quot;
+                }
+            </code-block>
+        </chapter>
+
+    </chapter>
+    <chapter title="Delete" id="delete">
+        <p>
+            To delete a record, use the <code>delete()</code> function:
+        </p>
+        <code-block lang="kotlin">
+            movie.delete()
+        </code-block>
+    </chapter>
+    <chapter title="Sort by" id="sort-by">
+        <chapter title="sortedBy" id="sortedBy">
+            <p>To sort results in ascending order, use <code>sortedBy</code>:</p>
+            <code-block lang="kotlin">
+                val movies = StarWarsFilm.all().sortedBy { it.sequelId }
+            </code-block>
+        </chapter>
+        <chapter title="sortedByDescending" id="sortedByDescending">
+            <p>To sort results in descending order, use <code>sortedByDescending</code>:</p>
+            <code-block lang="kotlin">
+                val movies = StarWarsFilm.all().sortedByDescending { it.sequelId }
+            </code-block>
+        </chapter>
+    </chapter>
```

Comment:
I feel like this chapter would be better placed as a sub-section in 'Create' chapter. Or maybe placed directly after 'Create'. But since the page is now much smaller, it's not hard to find here.

Reply:
Or perhaps even better if we place it in the 'Read' chapter since we have documented 'all()' there ?

Reply:
Oh yeah, that was a slow brain moment 😄  When I typed 'Create', I really actually meant 'Read'.